core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -179,9 +187,25 @@ public void run() {
   }
 
   private class TransactionRunner implements Runnable {
+    private final AtomicReference<RunnerState> runnerState =
+        new AtomicReference<>(new RunnerState(false, false));
+
+    private class RunnerState {
+      // used to signal a TransactionRunner to stop in the case where there are too many running
+      // i.e., the property for the pool size decreased and we have excess TransactionRunners
+      private final boolean stop;
+      // whether the TransactionRunner is executing the run() method (may or may not be working on
+      // a tx, could be waiting for a tx)
+      private final boolean isRunning;
```

Comment:
Wonder about changing this track exited which would be set in finally block to true.  Running can be false before it started and after its exited, just tracking its exited would distinguish between these cases.  Then if a thread was not stopped but it exited because of an exception we would know its finished.

```suggestion
      private final boolean exited;
```